manifests/pipeline/numalogic-prometheus-pipeline.yaml

```diff
@@ -105,7 +115,55 @@ spec:
               value: "6379"
             - name: REDIS_AUTH
               value: "5rmmfXtsbL"
-    - name: trainer
+    - name: trainer-1
```

Comment:
why 3 trainers?

Reply:
According to @whynowy , that would be a preferred design, rather than just passing a flag around (until a diamond flow is supported in numaflow)

Reply:
So these trainers are basically, one for main model, one for threshold and one for preprocess?

Reply:
No, they are the same exact, and they train all the models (preprocess, main, threshold) for a given composite key. Ideally, they should merge on a single trainer, but numaflow doesn't support that yet.

Reply:
No, we can still have just one trainer vertex if thats the case. since the trainer is same.

Reply:
No need of a flag too, since we are training all even with one request.

Reply:
Okay so as discussed offline, we shall be validating the payload type. If the payload type is of trainer payload, we shall just pass on, until it reaches the trainer. Otherwise, just process the payload.